ui/apps/dashboard/src/components/Insights/InsightsTabManager/InsightsTabManager.tsx

```diff
@@ -72,7 +72,12 @@ export function useInsightsTabManager(
         createTabBase(makeEmptyUnsavedQuery());
       },
       createTabFromQuery: (query: Query | QuerySnapshot | QueryTemplate) => {
-        if (isQueryTemplate(query) || isQuerySnapshot(query)) {
+        if (isQueryTemplate(query)) {
+          createTabBase({ ...makeEmptyUnsavedQuery(), query: query.query, name: query.name });
```

Comment:
maybe we want to add "From Template:" or something like that, but we can see what user feedback is first.

Also from playing around on preview, looks like it's fine if we have multiple tabs with the same name 👍